examples/binary_examples/PyFstat_example_binary_mcmc_vs_grid_comparison.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+
+import pyfstat
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+
+# Set to false to include eccentricity
+circular_orbit = False
+
+label = os.path.splitext(os.path.basename(__file__))[0]
+outdir = os.path.join("PyFstat_example_data", label)
+
+# Parameters to generate a data set
+data_parameters = {
+    "sqrtSX": 1e-22,
+    "tstart": 1000000000,
+    "duration": 60 * 86400,
+    "detectors": "H1,L1",
+    "Tsft": 1800,
+    "Band": 4,
+}
+
+# Injected signal parameters
+tend = data_parameters["tstart"] + data_parameters["duration"]
+mid_time = 0.5 * (data_parameters["tstart"] + tend)
+depth = 10.0
+signal_parameters = {
+    "tref": data_parameters["tstart"],
+    "F0": 100.0,
+    "F1": 0,
+    "F2": 0,
+    "Alpha": 0.5,
+    "Delta": 0.5,
+    "period": 5 * 24 * 3600.0,
+    "asini": 10.0,
+    "tp": mid_time * 1.05,
+    "argp": 0.0 if circular_orbit else 0.54,
+    "ecc": 0.0 if circular_orbit else 0.7,
+    "h0": data_parameters["sqrtSX"] / depth,
+    "cosi": 1.0,
+}
+
+
+print("Generating SFTs with injected signal...")
+writer = pyfstat.BinaryModulatedWriter(
+    label="simulated_signal",
+    outdir=outdir,
+    **data_parameters,
+    **signal_parameters,
+)
+writer.make_data()
+print("")
+
+print("Performing Grid Search...")
+
+# Create ad-hoc grid and compute Fstatistic around injection point
+half_points_per_dimension = 2
+search_keys = ["period", "asini", "tp"] + ["argp", "ecc"]
```

Comment:
should the second part be only on `not circular_orbit`?

Reply:
I'll polish that part a bit. I literally implemented the circular orbit as a "put ifs everywhere" (i.e. I didn't think about it in the beginning), so this can be done more elegantly.